ts-rs/src/lib.rs

```diff
@@ -380,39 +391,114 @@ pub trait TS {
+    /// `TS_RS_EXPORT_DIR` environment variable. By default, `./bindings` will be used.
+    /// 
+    /// To specify a target directory manually, use [`TS::export_to`].
+    ///
+    /// To alter the filename or path of the type within the target directory, 
+    /// use `#[ts(export_to = "...")]`.
     fn export() -> Result<(), ExportError>
     where
         Self: 'static,
     {
-        export::export_type_with_dependencies::<Self>()
-    }
-
-    /// Manually export this type to a file with a file with the specified path. This
-    /// function will ignore the `#[ts(export_to = "..)]` attribute.
+        let path = Self::default_output_path()
+            .ok_or_else(std::any::type_name::<Self>)
+            .map_err(ExportError::CannotBeExported)?;
+        
+        export::export_to::<Self, _>(path)
+    }
+    
+    /// Manually export this type to the given path. 
+    /// To export this type together with all of its dependencies, use [`TS::export_all`].
+    /// 
+    /// Unlike [`TS::export`], this function disregards `TS_RS_EXPORT_DIR` and 
+    /// `#[ts(export_to = "...")]`.
+    ///
+    /// # Automatic Exporting
+    /// Types annotated with `#[ts(export)]`, together with all of their dependencies, will be 
+    /// exported automatically whenever `cargo test` is run.  
+    /// In that case, there is no need to manually call this function.
+    ///
+    /// To alter the filename or path of the type within the target directory, 
+    /// use `#[ts(export_to = "...")]`.
     fn export_to(path: impl AsRef<Path>) -> Result<(), ExportError>
-    where
-        Self: 'static,
+        where
+            Self: 'static,
+    {
+        export::export_to::<Self, _>(path)
+    }
+    
+    /// Manually export this type to the filesystem, together with all of its dependencies.  
+    /// To export only this type, without its dependencies, use [`TS::export`].
+    ///
+    /// # Automatic Exporting
+    /// Types annotated with `#[ts(export)]`, together with all of their dependencies, will be 
+    /// exported automatically whenever `cargo test` is run.  
+    /// In that case, there is no need to manually call this function.
+    ///
+    /// # Target Directory
+    /// The target directory to which the types will be exported may be changed by setting the
+    /// `TS_RS_EXPORT_DIR` environment variable. By default, `./bindings` will be used.
+    /// 
+    /// To specify a target directory manually, use [`TS::export_all_to`].
+    /// 
+    /// To alter the filenames or paths of the types within the target directory, 
+    /// use `#[ts(export_to = "...")]`.
+    fn export_all() -> Result<(), ExportError>
+        where
+            Self: 'static,
+    {
+        export::export_all_into::<Self>(&*export::default_out_dir())
+    }
+    
+    /// Manually export this type into the given directory, together with all of its dependencies.  
+    /// To export only this type, without its dependencies, use [`TS::export`].
```

Comment:
Minor nitpick: I think this doc comment should link to `TS::export_to` instead of `TS::export` because it better corresponds to what is being described here (same behavior, but no dependencies)